include/boost/compute/detail/lru_cache.hpp

```diff
@@ -97,9 +97,7 @@ class lru_cache
             m_list.push_front(key);
 
             // update iterator in map
-            j = m_list.begin();
-            const value_type &value = i->second.first;
-            m_map[key] = std::make_pair(value, j);
+            i->second.second = m_list.begin();
 
             // return the value
             return value;
```

Comment:
Where is the value from.

Reply:
thanks, I forgot to update it. updated the PR now :)